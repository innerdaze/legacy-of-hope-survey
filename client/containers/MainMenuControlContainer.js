import { connect } from 'react-redux'
import { uiOperations } from '../features/ui'
import MainMenuControl from '../components/MainMenuControl'

const mapDispatchToProps = dispatch => {
  return {
    showMenu: () => dispatch(uiOperations.uiShowMenu())
  }
}

const MainMenuControlContainer = connect(null, mapDispatchToProps)(
  MainMenuControl
)

export default MainMenuControlContainer
