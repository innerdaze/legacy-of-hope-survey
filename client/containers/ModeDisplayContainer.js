import { connect } from 'react-redux'
import ModeDisplay from '../components/ModeDisplay'
import { orderSelectors } from '../features/order'
const mapStateToProps = state => ({
  mode: orderSelectors.modeSelector(state)
})

export default connect(mapStateToProps)(ModeDisplay)
