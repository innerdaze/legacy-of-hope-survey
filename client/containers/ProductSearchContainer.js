import { connect } from 'react-redux'
import ProductSearch from '../components/ProductSearch'
import { productOperations, productSelectors } from '../features/product'
import { orderOperations } from '../features/order'

export default connect(
  state => ({
    results: productSelectors.lastMatches(state)
  }),
  dispatch => ({
    search: query =>
      dispatch(
        productOperations.searchProducts(
          query,
          productOperations.searchProductByProductName
        )
      ),
    onSelect: product => dispatch(orderOperations.submitProduct(product))
  })
)(ProductSearch)
