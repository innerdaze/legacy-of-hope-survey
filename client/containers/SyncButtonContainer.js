import { connect } from 'react-redux'
import { syncOperations } from '../features/sync'
import SyncButton from '../components/SyncButton'

const mapDispatchToProps = dispatch => {
  return {
    sync: () => {
      dispatch(syncOperations.sync())
    }
  }
}

const SyncButtonContainer = connect(null, mapDispatchToProps)(SyncButton)

export default SyncButtonContainer
