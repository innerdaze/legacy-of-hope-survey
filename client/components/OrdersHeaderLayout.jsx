import React from 'react'
import Box from 'grommet/components/Box'
import Search from 'grommet/components/Search'
import Header from 'grommet/components/Header'
import MainMenuControlContainer from '../containers/MainMenuControlContainer'
import SyncButtonContainer from '../containers/SyncButtonContainer'
import ModeDisplayContainer from '../containers/ModeDisplayContainer'
import ProductSearchContainer from '../containers/ProductSearchContainer'

const OrdersHeaderLayout = () => (
  <Header>
    <MainMenuControlContainer/>
    <ModeDisplayContainer/>
    <Box
      flex
      fill
      direction='row'
      justify='end'
      responsive={false}
      >
      <SyncButtonContainer/>
      <ProductSearchContainer/>
    </Box>
  </Header>
)

export default OrdersHeaderLayout
