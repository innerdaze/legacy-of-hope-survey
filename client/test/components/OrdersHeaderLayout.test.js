import React from 'react'
import OrdersHeaderLayout from '../../components/OrdersHeaderLayout'
import MainMenuControlContainer from '../../containers/MainMenuControlContainer'
import ModeDisplayContainer from '../../containers/ModeDisplayContainer'
import SyncButtonContainer from '../../containers/SyncButtonContainer'
import ProductSearchContainer from '../../containers/ProductSearchContainer'

let wrapper

describe('Testing on OrdersHeaderLayout...', () => {
  beforeEach(() => {
    wrapper = shallow(<OrdersHeaderLayout />)
  })

  describe('OrdersHeaderLayout layout', () => {
    test('Expect OrdersHeaderLayout to have parent Header and a Box comp', () => {
      expect(wrapper.find('Header')).toHaveLength(1)

      expect(wrapper).toMatchSnapshot()
    })

    test('Expect Header comp to have MainMenuControlContainer and ModeDisplayContainer', () => {
      expect(
        wrapper.find('Header').contains(<MainMenuControlContainer />)
      ).toEqual(true)
      expect(wrapper.find('Header').contains(<ModeDisplayContainer />)).toEqual(
        true
      )

      expect(wrapper).toMatchSnapshot()
    })

    test('Expect Box comp to have SyncButtonContainer and ProductSearchContainer', () => {
      expect(
        wrapper
          .find('Header')
          .find('Box')
          .contains(<SyncButtonContainer />)
      ).toEqual(true)
      expect(
        wrapper
          .find('Header')
          .find('Box')
          .contains(<ProductSearchContainer />)
      ).toEqual(true)

      expect(wrapper).toMatchSnapshot()
    })
  })
})
