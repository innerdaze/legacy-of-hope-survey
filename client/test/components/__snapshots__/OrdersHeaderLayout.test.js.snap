// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on OrdersHeaderLayout... OrdersHeaderLayout layout Expect Box comp to have SyncButtonContainer and ProductSearchContainer 1`] = `
<Header
  align="center"
  direction="row"
  pad={
    Object {
      "between": "small",
      "horizontal": "none",
      "vertical": "none",
    }
  }
  responsive={false}
>
  <Connect(MainMenuControl) />
  <Connect(ModeDisplay) />
  <Box
    announce={false}
    direction="row"
    flex={true}
    focusable={true}
    justify="end"
    pad="none"
    responsive={false}
    tag="div"
  >
    <Connect(SyncButton) />
    <Connect(ProductSearch) />
  </Box>
</Header>
`;

exports[`Testing on OrdersHeaderLayout... OrdersHeaderLayout layout Expect Header comp to have MainMenuControlContainer and ModeDisplayContainer 1`] = `
<Header
  align="center"
  direction="row"
  pad={
    Object {
      "between": "small",
      "horizontal": "none",
      "vertical": "none",
    }
  }
  responsive={false}
>
  <Connect(MainMenuControl) />
  <Connect(ModeDisplay) />
  <Box
    announce={false}
    direction="row"
    flex={true}
    focusable={true}
    justify="end"
    pad="none"
    responsive={false}
    tag="div"
  >
    <Connect(SyncButton) />
    <Connect(ProductSearch) />
  </Box>
</Header>
`;

exports[`Testing on OrdersHeaderLayout... OrdersHeaderLayout layout Expect OrdersHeaderLayout to have parent Header and a Box comp 1`] = `
<Header
  align="center"
  direction="row"
  pad={
    Object {
      "between": "small",
      "horizontal": "none",
      "vertical": "none",
    }
  }
  responsive={false}
>
  <Connect(MainMenuControl) />
  <Connect(ModeDisplay) />
  <Box
    announce={false}
    direction="row"
    flex={true}
    focusable={true}
    justify="end"
    pad="none"
    responsive={false}
    tag="div"
  >
    <Connect(SyncButton) />
    <Connect(ProductSearch) />
  </Box>
</Header>
`;
